socketio_block.py

```diff
@@ -238,12 +243,20 @@ def handle_data(self, data):
 
     def _connect_to_socket(self):
         try:
+            # Don't need the job any more
+            self._connection_job = None
+
             self._do_handshake()
 
             url = self._get_ws_url()
 
             self._logger.debug("Connecting to %s" % url)
 
+            # In case the client is sticking around, close it before creating a
+            # new one
+            if self._client:
+                self._client.close_connection()
```

Comment:
where does this close_connection method come from? I don't see it here or in the ws4py docs